pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -347,6 +351,14 @@ stdenv.mkDerivation ({
 
     isHaskellLibrary = hasActiveLibrary;
 
+    # TODO: ask why the split outputs are configurable at all?
+    # TODO: include tests for split if possible
+    # Given the haskell package, returns
+    # the directory containing the haddock documentation.
+    # `null' if no haddock documentation was built.
+    # TODO: fetch the self from the fixpoint instead
+    haddockDir = self: if doHaddock then "${docdir self.doc}/html" else null;
```

Comment:
I’m not sure how this can be done. Something like

```nix
let getDocPath = drv: overrideCabal drv (drv: if drv.separateDocOutputPath then "$doc/share/doc" else "$out/share/doc"; );
```

cannot be used from within `generic-builder.nix` since there is no `self`, circumventing the idea that the string is only defined in one place.